### PR TITLE
ARROW-15917: [Java][Docs] Document how to use Flight artifacts

### PR DESCRIPTION
**Pull request by davisusanibar**

### CODE REVIEW

File: docs/source/java/install.rst [hunk 1]
```diff
@@ -28,15 +28,15 @@ Java modules are regularly built and tested on macOS and Linux distributions.
...
+Java modules are currently compatible with JDK 8, 9, 10, or 11, but only JDK 11 is tested in CI.
 
 Installing from Maven
 ---------------------
 
 By default, Maven will download from the central repository: https://repo.maven.apache.org/maven2/org/apache/arrow/
 
 Configure your pom.xml with the Java modules needed, for example:
-``arrow-memory-netty``, ``arrow-format``, and ``arrow-vector``.
+arrow-vector, arrow-memory-netty and flight-core.
```
Comment (lidavidm) [inline docs/source/java/install.rst:39]:
```suggestion
arrow-vector, arrow-memory-netty, and flight-core.
```

Comment (lidavidm) [reply docs/source/java/install.rst:39]: That said: I would rather we show a separate POM with the changes needed for Flight, including os-maven-plugin.

File: docs/source/java/install.rst [hunk 2]
```diff
@@ -82,85 +94,121 @@ Installing Nightly Packages
     These packages are not official releases. Use them at your own risk.
 
 Arrow nightly builds are posted on the mailing list at `builds@arrow.apache.org`_.
-The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be found at `Github Nightly`_.
+The artifacts are uploaded to GitHub. For example, for 2022/03/19, they can be found at `Github Nightly`_.
```
Comment (lidavidm) [inline docs/source/java/install.rst:97]: I don't think we need to change the nightly instructions for Flight. We just need to show using os-maven-plugin above.

File: docs/source/java/install.rst (hunk 1 above)
Comment (davisusanibar) [reply docs/source/java/install.rst]: Added

File: docs/source/java/install.rst (hunk 2 above)
Comment (lidavidm) [reply docs/source/java/install.rst]: Ah, I see, thanks.

File: docs/source/java/install.rst [hunk 3]
```diff
@@ -62,12 +62,43 @@ Configure your pom.xml with the Java modules needed, for example:
                 <artifactId>arrow-memory-netty</artifactId>
                 <version>${arrow.version}</version>
             </dependency>
+        </dependencies>
+    </project>
+
+In case to add Arrow Java Flight dependencies don't forget to
+add `os-maven-plugin` extension. This `os-maven-plugin` generate
+useful platform-dependent project properties such as `os.detected.name`
+or `os.detected.arch` needed by Arrow Java Flight dependencies.
```
Comment (lidavidm) [inline docs/source/java/install.rst:71]:
```suggestion
To use the Arrow Flight dependencies, also add the ``os-maven-plugin``
plugin. This plugin generates useful platform-dependent properties
such as ``os.detected.name`` and ``os.detected.arch`` needed to resolve
transitive dependencies of Flight.
```

File: docs/source/java/install.rst [hunk 4]
```diff
@@ -87,80 +118,116 @@ The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be f
...
-        <dependency>
-            <groupId>org.apache.arrow</groupId>
-            <artifactId>arrow-format</artifactId>
+            <artifactId>flight-core</artifactId>
             <version>${arrow.version}</version>
         </dependency>
     </dependencies>
 
-3. Download packages needed to a temporary directory:
+3. Download packages needed to a temporary directory: pom and jar files
```
Comment (lidavidm) [inline docs/source/java/install.rst:140]:
```suggestion
3. Download the necessary pom and jar files to a temporary directory:
```

File: docs/source/java/install.rst [hunk 5]
```diff
@@ -87,80 +118,116 @@ The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be f
...
+    ├── arrow-memory-core-8.0.0.dev254.jar
+    ├── arrow-memory-core-8.0.0.dev254.pom
+    ├── arrow-memory-netty-8.0.0.dev254.jar
+    ├── arrow-memory-netty-8.0.0.dev254.pom
+    ├── arrow-vector-8.0.0.dev254.jar
+    ├── arrow-vector-8.0.0.dev254.pom
+    ├── flight-core-8.0.0.dev254.jar
+    └── flight-core-8.0.0.dev254.pom
+
+4. Install the artifacts to the local Maven repository with **mvn install:install-file**:
```
Comment (lidavidm) [inline docs/source/java/install.rst:175]:
```suggestion
4. Install the artifacts to the local Maven repository with ``mvn install:install-file``:
```

File: docs/source/java/install.rst [hunk 6]
```diff
@@ -87,80 +118,116 @@ The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be f
...
+    $ mvn install:install-file -Dfile="$(pwd)/arrow-memory-netty-8.0.0.dev254.jar" -DgroupId=org.apache.arrow -DartifactId=arrow-memory-netty -Dversion=8.0.0.dev254 -Dpackaging=jar
+    $ mvn install:install-file -Dfile="$(pwd)/arrow-flight-8.0.0.dev254.pom" -DgroupId=org.apache.arrow -DartifactId=arrow-flight -Dversion=8.0.0.dev254 -Dpackaging=pom
+    $ mvn install:install-file -Dfile="$(pwd)/flight-core-8.0.0.dev254.pom" -DgroupId=org.apache.arrow -DartifactId=flight-core -Dversion=8.0.0.dev254 -Dpackaging=pom
+    $ mvn install:install-file -Dfile="$(pwd)/flight-core-8.0.0.dev254.jar" -DgroupId=org.apache.arrow -DartifactId=flight-core -Dversion=8.0.0.dev254 -Dpackaging=jar
 
 5. Validate that the packages were installed:
 
 .. code-block:: shell
 
     $ tree /Users/arrow/.m2/repository/org/apache/arrow
```
Comment (lidavidm) [inline docs/source/java/install.rst:197]:
```suggestion
    $ tree ~/.m2/repository/org/apache/arrow
```

File: docs/source/java/install.rst [hunk 7]
```diff
@@ -87,80 +118,116 @@ The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be f
...
+    │   ├── 8.0.0.dev254
+    │   │   ├── _remote.repositories
+    │   │   ├── arrow-vector-8.0.0.dev254.jar
+    │   │   └── arrow-vector-8.0.0.dev254.pom
+    └── flight-core
+        ├── 8.0.0.dev254
+        │   ├── flight-core-8.0.0.dev254.jar
+        │   └── flight-core-8.0.0.dev254.pom
+
+6. Compile your project like usual with **mvn clean install**.
```
Comment (lidavidm) [inline docs/source/java/install.rst:230]:
```suggestion
6. Compile your project like usual with ``mvn clean install``.
```

**Review by lidavidm — APPROVED** [review]

**Pull request closed without merging**